crates/matrix-sdk-common/src/ring_buffer.rs

```diff
@@ -52,6 +55,19 @@ impl<T> RingBuffer<T> {
     pub fn iter(&self) -> Iter<'_, T> {
         self.inner.iter()
     }
+
+    pub fn clear(&mut self) {
+        self.inner.clear();
+    }
+
+    pub fn extend<I>(&mut self, iter: I)
+    where
+        I: IntoIterator<Item = T>,
+    {
+        for item in iter.into_iter() {
+            self.push(item);
```

Comment:
I looks for a better way, but because `truncate` on `VecDeque` deletes from the end and I didn't see one for the beginning, I wasn't convinced I could do much better.

For my use, which is very small buffers, this should be fine, but I'd welcome suggestions.

Reply:
I think for a general `Iterator` based version, this is as good as it gets. But this should probably be an implementation of the `Extend` trait.

We could also provide something potentially more efficient:

```rust
pub fn extend_exact<I>(&mut self, into_iter: I)
where
    I: IntoIterator<Item = T>,
    I::IntoIter: ExactSizeIterator,
{
    let iter = into_iter.into_iter();
    // use `self.inner.drain(..some_number)` if the iterator has more items
    // than we have capacity remaining
    // skip `self.inner.capacity() - iter.len()` items if that number is positive
    self.inner.extend(iter);
}
```

Reply:
I moved `extend` into the `Extend` trait - feel free to merge when ready.